lua/telescope/builtin/lsp.lua

```diff
@@ -6,9 +6,33 @@ local utils = require('telescope.utils')
 
 local conf = require('telescope.config').values
 
+local function check_capabilities(feature)
+  local clients = vim.lsp.buf_get_clients(0)
+
+  local supported_client = false
+  for _, client in pairs(clients) do
+    supported_client = client.resolved_capabilities[feature]
+    if supported_client then goto continue end
+  end
+
+  ::continue::
+  if supported_client then
+    return true
+  else
+    if #clients == 0 then
+      print("LSP: no client attached")
+    else
+      print("LSP: server does not support " .. feature)
+    end
+    return false
+  end
+end
+
 local lsp = {}
 
 lsp.references = function(opts)
+  if not check_capabilities("find_references") then return end
```

Comment:
There is an apply_checks function at the bottom of the file which will wrap all functions that will be exported. So you can do this only once down there

Reply:
I think that's taken care of. I had to create a new table to map the function names to the actual parameter name used in the capability check, rather than renaming the functions and changing peoples mappings.